templates/bake/Template/index.twig

```diff
@@ -16,7 +16,7 @@
 <?php
 /**
  * @var \{{ namespace }}\View\AppView $this
- * @var \{{ entityClass }}[]|\Cake\Collection\CollectionInterface ${{ pluralVar }}
+ * @var array<\{{ entityClass }}>|\Cake\Collection\CollectionInterface<\{{ entityClass }}> ${{ pluralVar }}
```

Comment:
As seen [here](https://github.com/cakephp/cakephp/pull/16651/files#diff-beab97e6f7fcea692dab3a2b6ce9b208aca80beb7490820345e88a28f9fb7c39R2181) phpcs doesn't like the item/class type interfaces even if they extend an iterable type.

Reply:
Refs https://github.com/cakephp/cakephp/pull/16651#pullrequestreview-1053719318